### PR TITLE
[sb/eh] mypy

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def __init__(self, client: "EventHubConsumerClient", source: str, **kwargs: Any)
         link_properties: Dict[bytes, int] = {}
         self._error = None
         self._timeout = 0
-        self._idle_timeout = (idle_timeout * self._amqp_transport.TIMEOUT_FACTOR) if idle_timeout else None
+        self._idle_timeout: Optional[int] = (idle_timeout * self._amqp_transport.TIMEOUT_FACTOR) if idle_timeout else None
```
Comment (kashifkhan) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py:107]: have we used float for timeout or this one always int?

Comment (kashifkhan) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py:107]: just had to scroll a bit, its always been int. Have we been uniform about this?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_uamqp_transport.py [hunk 2]
```diff
@@ -658,7 +661,7 @@ def create_receive_client(receiver: "ServiceBusReceiver", **kwargs: Any) -> "Rec
 
         @staticmethod
         def on_attach(  # pylint: disable=unused-argument
-            receiver: "ServiceBusReceiver",
+            receiver: Union["ServiceBusReceiver", "ServiceBusReceiverAsync"],
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_uamqp_transport.py:664]: do both of these need be under a typing guard instead?

**Review by swathipil — APPROVED** [review]

File: sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py (hunk 1 above)
Comment (l0lawrence) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py]: nope its float someplaces and int elsewhere

**Review by kashifkhan — APPROVED** [review]

File: sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_uamqp_transport.py (hunk 2 above)
Comment (l0lawrence) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_uamqp_transport.py:664]: Isn't a type guard basically a bool, why would we need that here?

Comment (kashifkhan) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_uamqp_transport.py:664]: yep, I was mistake, was thinking of something else.

**Pull request merged**